aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -1061,194 +1062,152 @@ module aptos_framework::stake {
     /// pending inactive validators so they no longer can vote.
     /// 4. The validator's voting power in the validator set is updated to be the corresponding staking pool's voting
     /// power.
-    public(friend) fun on_new_epoch() acquires StakePool, AptosCoinCapabilities, ValidatorConfig, ValidatorPerformance, ValidatorSet, ValidatorFees {
+    /// 5. Return the new validator set.
+    ///
+    /// If `commit` is false, still do the calculation but prevent any resource update.
+    public(friend) fun update_validator_set_on_new_epoch(commit: bool): ValidatorSet acquires StakePool, AptosCoinCapabilities, ValidatorConfig, ValidatorPerformance, ValidatorSet, ValidatorFees {
         let validator_set = borrow_global_mut<ValidatorSet>(@aptos_framework);
         let config = staking_config::get();
         let validator_perf = borrow_global_mut<ValidatorPerformance>(@aptos_framework);
 
+        let new_stakes_by_validator = simple_map::new<address, u64>();
```

Comment:
nit: < address, u64 > is not required as that can be inferred from the add call below